server/lib/chart.js

```diff
@@ -0,0 +1,127 @@
+const _ = require('lodash');
+const util = require('./util');
+
+module.exports = {
+  barChar,
+};
+
+/**
+ * @function barChar
+ * This function help to skip some manipulation for respecting chartjs lib format...
+ */
+
+function barChar(params) {
+  const {
+    label, data, item, yAxesLabelString,
+    xAxesLabelString, canvasId,
+  } = params;
+
+  const chart = {
+    labels : [],
+    data : [],
+  };
+  const serieMap = {};
+
+  const dailyData = _.groupBy(data, label);
+  chart.labels = Object.keys(dailyData).sort((a, b) => {
+    return a.localeCompare(b);
+  });
+
+  const items = Object.keys(_.groupBy(data, item.uuid));
+  chart.labels.forEach(day => {
```

Comment:
What is this loop doing?  I've been reading this function over and over and I can't figure it out.